File: typescript/tracing/langchain/03_chat_model_batch.ts
import { fakeChat, initRespan, messageText, shutdown, tracingConfig } from "./_shared";

export async function chatModelBatch(): Promise<void> {
  const runtime = await initRespan("typescript-langchain-chat-model-batch");
  const model = fakeChat(["first batch response", "second batch response"]);

  try {
    const responses = await model.batch(
      ["First request", "Second request"],
      tracingConfig(runtime, "chat_model_batch"),
    );
    console.log(responses.map(messageText).join(" | "));
  } finally {
    await shutdown(runtime);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  await chatModelBatch();
}

File: python/tracing/langchain/03_chat_model_batch.py
"""Chat model batch."""

from langchain_core.language_models.fake_chat_models import FakeListChatModel

from _shared import init_telemetry, message_text, tracing_config


def chat_model_batch() -> None:
    telemetry = init_telemetry("langchain-chat-model-batch")
    model = FakeListChatModel(responses=["alpha", "beta", "gamma"])
    responses = model.batch(
        ["Return alpha.", "Return beta.", "Return gamma."],
        config=tracing_config("chat_model_batch", {"batch_size": 3}),
    )
    print([message_text(response) for response in responses])
if __name__ == "__main__":
    chat_model_batch()
